services/wallet/activity/service.go

```diff
@@ -109,6 +135,40 @@ func (s *Service) GetTxDetails(ctx context.Context, id string) (*EntryDetails, e
 	return getTxDetails(ctx, s.db, id)
 }
 
+func (s *Service) GetLatestBlockNumber(ctx context.Context, chainID uint64) (uint64, error) {
+	s.blkMu.Lock()
```

Comment:
personal suggestion, I usually like to use only `[R]Lock()` and `defer [R]Unlock()`, so I wrap access to lock-protected resources in their own function.
In this case, getting or setting latestBlockNumbers should be their own function.